app/views/bands/index.html.erb

```diff
@@ -0,0 +1,4 @@
+<% @bands.each do |band| %>
+    <h2> <%= band.name  %> </h2>
+     <%= link_to "", band_path(band[:id]) %>
```

Comment:
you can passe the object instead of the id, and band.name should be the link =>
<%= link_to band.name, band_path(band) %>